opm/parser/eclipse/EclipseState/Schedule/OilVaporizationProperties.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef DRSDT_HPP
+#define DRSDT_HPP
+#include <iostream>
+#include <string>
+#include <memory>
+#include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
+
+namespace Opm
+{
+    class OilVaporizationProperties
+    {
+    public:
+        OilVaporizationProperties();
+
+        static std::shared_ptr<OilVaporizationProperties> createOilVaporizationPropertiesDRSDT(double maximum, std::string option);
+        static std::shared_ptr<OilVaporizationProperties> createOilVaporizationPropertiesDRVDT(double maximum);
+        static std::shared_ptr<OilVaporizationProperties> createOilVaporizationPropertiesVAPPARS(double vap1, double vap2);
+        Opm::OilVaporizationEnum getType() const;
+        double getVap1() const;
+        double getVap2() const;
+        double getMaximum() const;
+        std::string getOption() const;
+
+    private:
+        Opm::OilVaporizationEnum m_type;
+        double m_maximum;
+        double m_vap1;
+        double m_vap2;
+        std::string m_option;
```

Comment:
This string can only take the values `ALL` and `FREE` and should be internalized as:
```C++
bool m_maxDRSDT_allCells;
```